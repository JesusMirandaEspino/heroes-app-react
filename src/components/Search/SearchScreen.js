import React from 'react'
import { heroes } from '../../data/heroes';
import { HeroesCard } from '../Heroes/HeroesCard';

export const SearchScreen = () => {

    const heroesFiltered = heroes;


    return (
        <div>
            <h1>SearchScreen</h1>
            <hr/>


            <div className="row">

                <div className="col-4" >
                    <h4> SearchForm </h4>
                    <hr/>

                    <form>

                        <label> Find your Hero </label>
                        <input type="text"  placeholder="Hero"  className="form-control" />

                        <button  type="submit" className="btn m-1 btn-block btn-outline-primary" >Search</button>

                    </form>

                </div>

                <div className="col-8" >

                    <h4>Results</h4>
                    <hr/>

                    {
                        heroesFiltered.map(
                            hero => ( <HeroesCard  key={ hero.id }  { ...hero } /> )
                        )
                    }
                    
                </div>

            </div>


        </div>
    )
}
