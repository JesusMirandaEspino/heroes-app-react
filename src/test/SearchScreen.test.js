import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter, Route } from 'react-router-dom';

import { SearchScreen } from '../components/Search/SearchScreen';

describe( ' Pruebas con <SearchScreen /> ', () => {

    test('Debe de mostrarse con los valores por defecto correctamente ', () => {

        const wrapper = mount(
            <MemoryRouter initialEntries={ ['/search'] } >
                <Route path='/search' component={SearchScreen} />
            </MemoryRouter>
        );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find( '.textinfo' ).text().trim() ).toBe('Search a Hero');

    });


    test('debe de mostrar a Batman y el input con el valor del QueryString  ', () => {
        
        const wrapper = mount(
            <MemoryRouter initialEntries={ ['/search?q=batman'] } >
                <Route path='/search' component={SearchScreen} />
            </MemoryRouter>
        );

            expect( wrapper.find('input').prop('value') ).toBe('batman');
            expect( wrapper ).toMatchSnapshot();

    });


    test('Debe mostrar un error si no encuntran al heroe ', () => {

        const wrapper = mount(
            <MemoryRouter initialEntries={ ['/search?q=batman123'] } >
                <Route path='/search' component={SearchScreen} />
            </MemoryRouter>
        );

            expect( wrapper.find('.textdanger').text().trim() ).toBe(`There is no a Hero batman123`);
            expect( wrapper ).toMatchSnapshot();    

    });
    

    test('Debe de llamar el PUSH del history ', () => {
        
        const history = {
            push: jest.fn()
        };

        const wrapper = mount(
            <MemoryRouter initialEntries={ ['/search?q=batman123'] } >
                <Route path='/search'  component={ () => <SearchScreen  history={ history } />} />
            </MemoryRouter>
        );

        wrapper.find('input').simulate( 'change', {
            target: {
                name: 'searchText',
                value: 'batman'
            }
        });

        wrapper.find('form').prop('onSubmit')({
            preventDefault(){}
        });

        expect( history.push ).toHaveBeenCalledWith(`?q=batman `);

    });
    
    


});