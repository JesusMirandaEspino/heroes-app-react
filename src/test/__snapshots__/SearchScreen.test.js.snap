// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Pruebas con <SearchScreen />  Debe de mostrarse con los valores por defecto correctamente  1`] = `
<div>
  <h1>
    SearchScreen
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-4"
    >
      <h4>
         SearchForm 
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <label>
           Find your Hero 
        </label>
        <input
          className="form-control"
          name="searchText"
          onChange={[Function]}
          placeholder="Hero"
          type="text"
          value=""
        />
        <button
          className="btn m-1 btn-block btn-outline-primary"
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
    <div
      className="col-8"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        className="alert alert-info"
      >
        <p
          className="textinfo"
        >
           Search a Hero 
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[` Pruebas con <SearchScreen />  Debe mostrar un error si no encuntran al heroe  1`] = `
<div>
  <h1>
    SearchScreen
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-4"
    >
      <h4>
         SearchForm 
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <label>
           Find your Hero 
        </label>
        <input
          className="form-control"
          name="searchText"
          onChange={[Function]}
          placeholder="Hero"
          type="text"
          value="batman123"
        />
        <button
          className="btn m-1 btn-block btn-outline-primary"
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
    <div
      className="col-8"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        className="alert alert-danger"
      >
        <p
          className="textdanger"
        >
           There is no a Hero 
          batman123
           
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[` Pruebas con <SearchScreen />  debe de mostrar a Batman y el input con el valor del QueryString   1`] = `
<div>
  <h1>
    SearchScreen
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-4"
    >
      <h4>
         SearchForm 
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <label>
           Find your Hero 
        </label>
        <input
          className="form-control"
          name="searchText"
          onChange={[Function]}
          placeholder="Hero"
          type="text"
          value="batman"
        />
        <button
          className="btn m-1 btn-block btn-outline-primary"
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
    <div
      className="col-8"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        className=" card ms-3  animate__animated animate__fadeIn "
        style={
          Object {
            "maxWidth": 540,
          }
        }
      >
        <div
          className="row no-gutters "
        >
          <div
            className="col-md-4"
          >
            <img
              alt="Batman"
              className="card-img"
              src="./assets/img/dc-batman.jpg"
            />
          </div>
          <div
            className="col-md-8"
          >
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                 
                Batman
                 
              </h5>
              <p
                className="card-text"
              >
                 
                Bruce Wayne
                 
              </p>
              <p
                className="card-text"
              >
                <small
                  className="text-muted"
                >
                   
                  Detective Comics #27
                   
                </small>
              </p>
              <a
                href="/hero/dc-batman"
                onClick={[Function]}
              >
                 Saber mas... 
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
